images/builders/mlflow/run.sh

```diff
@@ -1,4 +1,57 @@
 #!/bin/sh
 
-mkdir -p .fuseml
-cp -rv ${FUSEML_FILES}/* .fuseml/ | awk '/Dockerfile/ {print $3}' | sed -e "s/[\r\n']//g"
+set -o pipefail
+
+# generates a checksum based on the contents of a conda.yaml file
+get_tag() {
+    file=${1:-"conda.yaml"}
+    dep_provider=""
+
+    while IFS= read -r line; do
+        name=$(echo ${line} | sed 's/ *- //g')
+        case $line in
+            '  - '*':')
+                dep_provider=$(printf '%s' "${name}" | tr -d :)
+            ;;
+            '  - '*)
+                if [ -z "${dependencies}" ]; then
+                    dependencies=${name}
+                else
+                    dependencies=${dependencies}'\n'${name}
+                fi
+            ;;
+            '    - '*)
+                dependencies=${dependencies}'\n'${dep_provider}'.'${name}
+            ;;
+        esac
+    done <"${file}"
+
+    printf "$dependencies" | sort | cksum | cut -f 1 -d ' '
+}
+
+conda_file="conda.yaml"
+
+if [ ! -f "${conda_file}" ]; then
+    echo "${conda_file} not found in $(pwd)"
+    exit 1
+fi
+
+registry=${FUSEML_REGISTRY:-"registry.fuseml-registry"}
+repository=${FUSEML_REPOSITORY:-"mlflow/trainer"}
+tag=$(get_tag)
+
+# destination is the task output, when using the internal FuseML registry we need to reference the repository
+# using the localhost address (see https://github.com/fuseml/fuseml/issues/65).
+destination="${FUSEML_REGISTRY:-127.0.0.1:30500}/${repository}:${tag}"
```

Comment:
I would say it's better to make the target registry a command line argument for this image, but that wouldn't solve the double identity problem, so I'll just go back to cringing in silence at the sight of these hard-coded hostnames and port numbers :)

Reply:
You are right, those hard coded values are bad, but needed until the registry is fixed.

You can also overwrite the registry through environment variables, or on the pipeline with `registry` as input.